### PR TITLE
New Device Support - ICT Digital Power Supply

### PR DESCRIPTION
**Pull request by EnzoZafra**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/discovery/sensors/current/ict-psu.inc.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+// Output Current
+// SNMPv2-SMI::enterprises.39145.11.8.0 = STRING: "0.4" -- outputCurrent
+
+$outputCurrent_oid = '.1.3.6.1.4.1.39145.11.8.0';
+$outputCurrent = trim(snmp_get($device, $outputCurrent_oid, '-Oqv'), '" ');
+if (!empty($outputCurrent)) {
+    $divisor = 1;
+    $index = '11.8.0';
```
Comment (laf) [inline includes/discovery/sensors/current/ict-psu.inc.php]: `$index` can just be `0` in this instance.

File: includes/discovery/sensors/voltage/ict-psu.inc.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+// Input Voltage
+// SNMPv2-SMI::enterprises.39145.11.6.0 = STRING: "120" -- inputVoltage
+
+$inputVoltage_oid = '.1.3.6.1.4.1.39145.11.6.0';
+$inputVoltage = trim(snmp_get($device, $inputVoltage_oid, '-Oqv'), '" ');
+if (!empty($inputVoltage)) {
+    $divisor = 1;
+    $index = '11.6.0';
```
Comment (laf) [inline includes/discovery/sensors/voltage/ict-psu.inc.php]: `$index` You can use `0` and `1` here as you know how many you have, the alternative is to use the OID name if one is available from the mib like `InputVoltage`

File: includes/discovery/sensors/voltage/ict-psu.inc.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// Output Voltage
+// SNMPv2-SMI::enterprises.39145.11.7.0 = STRING: "55.2" -- outputVoltage
+
+$outputVoltage_oid = '.1.3.6.1.4.1.39145.11.7.0';
+$outputVoltage = trim(snmp_get($device, $outputVoltage_oid, '-Oqv'), '" ');
+if (!empty($outputVoltage)) {
+    $divisor = 1;
+    $index = '11.7.0';
```
Comment (laf) [inline includes/discovery/sensors/voltage/ict-psu.inc.php]: Same as last comment.

File: includes/discovery/sensors/current/ict-psu.inc.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 Lorenzo Zafra
+ * @author     Lorenzo Zafra<zafra@ualberta.ca>
+ */
+
+// Output Current
+// SNMPv2-SMI::enterprises.39145.11.8.0 = STRING: "0.4" -- outputCurrent
+
+$outputCurrent_oid = '.1.3.6.1.4.1.39145.11.8.0';
```
Comment (laf) [inline includes/discovery/sensors/current/ict-psu.inc.php]: You've used the numeric OID all over but have provided a mib, it's much cleaner to use the name from the MIB if you can.

**Comment by EnzoZafra** [comment]
Hey @laf, I fixed the code upon your requests but it still says 'Changes requested'. Not sure if I did anything wrong

**Comment by murrant** [comment]
@EnzoZafra laf needs to manually change his review to make that go away ;)

**Pull request merged**